skimage/filters/ridges.py

```diff
@@ -105,9 +107,139 @@ def _check_sigmas(sigmas):
+    # order [d^2/dx^2, d^2/(dx*dy), ...], so in reverse order.
+    for deriv_dirs in itertools.combinations_with_replacement(idx, 2):
+        # E.g., for idx=[0, 1] we get deriv_dirs=[0, 0]; [0, 1]; [1, 1]
+
+        deriv_step1 = 1 * (idx == deriv_dirs[0])
+        deriv_step2 = 1 * (idx == deriv_dirs[1])
+        # E.g., for deriv_dirs=[0, 0] we get deriv_step1=[1, 0]
+        #                                and deriv_step2=[1, 0]
+        #       for deriv_dirs=[0, 1] we get deriv_step1=[1, 0]
+        #                                and deriv_step2=[0, 1]
+        #       etc., expressing the two successive derivative
+        #             operations in the Hessian
+
+        if order == 'rc':
+            deriv_step1 = deriv_step1[::-1]
+            deriv_step2 = deriv_step2[::-1]
+            # For, e.g., deriv_order=[2, 0], we want the second
+            # derivative in the "horizontal"/"row" direction, and
+            # just Gaussian smoothing in the "vertical"/"column"
+            # direction. To do that on an array, we need to
+            # differentiate as [0, 2], because the first direction
+            # is the vertical direction, and the second the horizontal
+            # direction. Hence, we reverse the list order.
+
+        # Apply two successive Gaussian filter operations, as per detailed in
+        # https://dsp.stackexchange.com/questions/78280/are-scipy-second-order-gaussian-derivatives-correct
+        if np.all(np.array(sigma)>1):
+            H_elems.append(
+                ndi.gaussian_filter(
+                    ndi.gaussian_filter(image, sigma=np.sqrt(1 / 2) * sigma,
+                                        mode=mode, cval=cval,
+                                        order=deriv_step1, truncate=8),
+                    sigma=np.sqrt(1 / 2) * sigma, mode=mode,
+                    cval=cval, order=deriv_step2,
+                    truncate=8)
+            )
+        else:
+            # For small values of sigma, the scipy Gaussian filter
+            # suffers from aliasing and edge artifacts, given that
+            # the filter will approximate a sinc or sinc derivative
+            # which only goes to 0 very slowly (order 1/n^2). Thus,
+            # we will use a much larger truncate value to reduce any
+            # edge artifacts.
+            H_elems.append(
+                ndi.gaussian_filter(
+                    ndi.gaussian_filter(image, sigma=np.sqrt(1 / 2) * sigma,
+                                        mode=mode, cval=cval,
+                                        order=deriv_step1, truncate=100),
+                    sigma=np.sqrt(1 / 2) * sigma, mode=mode,
+                    cval=cval, order=deriv_step2,
+                    truncate=100)
+            )
+            # The truncation must be fairly large here, as the FFT-based
+            # kernel does not go to 0 quickly, if a small sigma is
+            # chosen.
```

Comment:
You may define `truncate` outside the loop once for all as
```python
truncate = 8 if np.all(np.asarray(sigma) > 1) else 100
```
and save some lines of code 
```suggestion
        H_elems.append(
            ndi.gaussian_filter(
                ndi.gaussian_filter(image, sigma=np.sqrt(1 / 2) * sigma,
                                    mode=mode, cval=cval,
                                    order=deriv_step1, truncate=truncate),
                sigma=np.sqrt(1 / 2) * sigma, mode=mode,
                cval=cval, order=deriv_step2,
                truncate=truncate)
        )
```
you may also define the filtering kernel once for all outside the loop
```python
kernel = np.sqrt(1 / 2) * sigma
```
and this will raise an error if `sigma` is not an `ndarray` :wink: